pkg/reconcilers/base_reconciler.go

```diff
@@ -92,6 +93,41 @@ func (b *BaseReconciler) EventRecorder() record.EventRecorder {
 	return b.recorder
 }
 
+// ReconcileOptions contains options for reconcile objects.
+type ReconcileOptions struct {
+	// Ensure defaults and webhook mutations are considered before comparing
+	// existing vs desired state of resources when reconciling updates
+	// by dry-running the updates before
+	DryRunFirst *bool
+
+	GetOptions []client.GetOption
+
+	CreateOptions []client.CreateOption
+
+	DeleteOptions []client.DeleteOption
+
+	UpdateOptions []client.UpdateOption
+}
+
+// DryRunFirst sets the "dry run first" option to "true", ensuring all
+// validation, mutations etc first without persisting the change to storage.
+var DryRunFirst = dryRunFirst{}
+
+type dryRunFirst struct{}
```

Comment:
Maybe we could define `dryRunFirst` as a bool to make it explicit? we could still attach methods to it or are we planning to expand the props of it?

Reply:
it does not need to be a bool.  When you add the  `ReconcileOption` interface implementation called `DryRunFirst`, you are adding the following configuration option. 

```go
opts.DryRunFirst = ptr.To(true)
```

The bool field exists in the `ReconcileOptions` struct